### ISSUE TITLE
Use after free in UdHubListener

### ISSUE THREAD

**Issue by ntonnaett** [jacktrip/jacktrip#286]
scan-build found a use after free:
```
[40/59] Compiling C++ object jacktrip.p/src_UdpHubListener.cpp.o
../../../src/UdpHubListener.cpp:411:5: warning: Use of memory after it is freed [cplusplus.NewDelete]
    clientConnection->read(port_buf, size);
```

```
    // Get our port.
    qint32 udp_port;
    size = sizeof(udp_port);
    char *port_buf = new char[size];
    clientConnection->read(port_buf, size);
    udp_port = qFromLittleEndian<qint32>(port_buf);
    delete[] port_buf;
    
    // Then our jack client name.
    char name_buf[gMaxRemoteNameLength];
    clientConnection->read(name_buf, gMaxRemoteNameLength);
    clientName = QString::fromUtf8((const char *)name_buf);
    
    // We can discard our username and password length since we already have them.
    clientConnection->read(port_buf, size);
    clientConnection->read(port_buf, size);
```

Looks like it doesn't really want to read. Maybe something like a skip method would be good?

**Comment by psiborg112** [comment]
Oops. Had to change the code to use dynamic allocation for MSVC and it looks like I screwed up the change. Fixing now.

**Comment by psiborg112** [comment]
Fixed in #287. Good catch. (I managed to hose my certificate and key when cleaning out my build directory - stupid place to store them, I know - so I didn't properly test that change against the auth code.)

**Comment by ntonnaett** [comment]
Are we already able to deploy a Windows installer and MacOS App file? Maybe we could have that as a goal for this release too? Let's talk about it somewhere else.
